go/ql/lib/semmle/go/frameworks/Logrus.qll

```diff
@@ -49,4 +54,67 @@ module Logrus {
 
     override int getFirstFormattedParameterIndex() { result = argOffset + 1 }
   }
+
+  private class SetFormatterFunction extends Function {
+    SetFormatterFunction() {
+      this.hasQualifiedName(packagePath(), "SetFormatter") or
+      this.(Method).hasQualifiedName(packagePath(), "Logger", "SetFormatter")
+    }
+  }
+
+  private class LoggerFormatter extends Field {
+    LoggerFormatter() { this.hasQualifiedName(packagePath(), "Logger", "Formatter") }
+  }
+
+  private class JsonFormatter extends Type {
+    JsonFormatter() { this.hasQualifiedName(packagePath(), "JSONFormatter") }
+  }
+
+  /**
+   * Gets the types corresponding to sanitizing formatters.
+   */
+  private Type sanitizingFormatter() { result instanceof JsonFormatter }
+
+  /**
+   * An assignment statement that assigns a value to the `Formatter` property of a `Logger` object.
+   */
+  private class SetFormatterAssignment extends AssignStmt {
+    SetFormatterAssignment() {
+      exists(LoggerFormatter field | this.getAnLhs().(SelectorExpr).getSelector().refersTo(field))
+    }
+  }
+
+  /**
+   * Holds if there is local data flow to `node` that, at some point, has a sanitizing formatter
+   * type.
+   */
+  private predicate isSanitizingFormatter(DataFlow::Node node) {
```

Comment:
This predicate name doesn't quite reflect what it does. How about `mayBeSanitizingFormatter`?